Assets/Scripts/Framework/Buttons/BasicButton.cs

```diff
@@ -0,0 +1,25 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace ProefExamen.Framework.Buttons
+{
+    /// <summary>
+    /// An abstract class that is responsible for handling the button click event.
+    /// </summary>
+    [RequireComponent(typeof(Button))]
+    public abstract class BasicButton : MonoBehaviour
+    {
+        private Button _button;
+
+        private void Awake() => _button = GetComponent<Button>();
+
+        private void OnEnable() => _button.onClick.AddListener(OnButtonPressed);
+
+        private void OnDisable() => _button.onClick.RemoveListener(OnButtonPressed);
+
+        /// <summary>
+        /// The method that is called when the button is pressed.
+        /// </summary>
+        public abstract void OnButtonPressed();
```

Comment:
This abstract method needs to be `protected` right? Otherwise other classes could access this class and call OnButtonPressed while the button was not touched at all. It should be hidden from the public API.

Reply:
I agree. I will change it to `private protected`.